File: source_funcs/go_transcoder/PROGRAM_FIND_SMALLEST_DIFFERENCE_ANGLES_TWO_PARTS_GIVEN_CIRCLE/PROGRAM_FIND_SMALLEST_DIFFERENCE_ANGLES_TWO_PARTS_GIVEN_CIRCLE.go
package main

import "math"

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}

func f_gold(arr []int, n int) int {
	var (
		l   int = 0
		sum int = 0
		ans int = 360
	)
	for i := int(0); i < n; i++ {
		sum += arr[i]
		for sum >= 180 {
			ans = min(ans, int(math.Abs(float64(180-sum))*2))
			sum -= arr[l]
			l++
		}
		ans = min(ans, int(math.Abs(float64(180-sum))*2))
	}
	return ans
}

File: rust_funcs/cpp_transcoder/PROGRAM_FIND_SMALLEST_DIFFERENCE_ANGLES_TWO_PARTS_GIVEN_CIRCLE/PROGRAM_FIND_SMALLEST_DIFFERENCE_ANGLES_TWO_PARTS_GIVEN_CIRCLE.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}


fn f_gold(arr: &[f32;2], n: i32) -> i32 {
let mut l = 0; 
let mut sum = 0f32;  
let mut ans = 360f32;
for i in 0..n {
    sum += arr[i as usize];
    while sum >= 180f32 {
        ans = ans.min(2f32 * (180f32 - sum).abs());
        sum -= arr[l];
        l += 1;
    } 
    ans = ans.min(2f32 * (180f32 - sum).abs());
}
ans
}
